### ISSUE TITLE
DGB p2p connection disconnects every 60 seconds, no block updates received

### ISSUE THREAD

**Issue by fship7** [zone117x/node-open-mining-portal#626]
A disconnect happens between NOMP and the digibyted p2p port after exactly 60 seconds. No block updates are ever received via this method from digibyted. RPC polls and blockupdate methods both work fine. The peermagic supplied in the coin configs is correct. Connections are successfully made, and a peer shows up as connected using the digibyted api, but then a disconnect happens after 60 seconds every time. NOMP reconnects, and then another disconnect after 60 seconds. Repeat.

Could something cause this on the digibyted side? 
some logs:

--- nomp ----
2018-08-04 18:52:23 [Pool]      [digibyte-qubit] (Thread 2) p2p connection successful
2018-08-04 18:52:50 [Pool]      [digibyte-qubit] (Thread 1) Block notification via RPC polling
2018-08-04 18:52:50 [Pool]      [digibyte-qubit] (Thread 2) Block notification via RPC polling
2018-08-04 18:53:01 [Pool]      [digibyte-qubit] (Thread 1) Block notification via RPC polling
2018-08-04 18:53:01 [Pool]      [digibyte-qubit] (Thread 2) Block notification via RPC polling
2018-08-04 18:53:06 [Pool]      [digibyte-qubit] (Thread 2) Block notification via RPC polling
2018-08-04 18:53:06 [Pool]      [digibyte-qubit] (Thread 1) Block notification via RPC polling
2018-08-04 18:53:20 [Pool]      [digibyte-qubit] (Thread 2) Block notification via RPC polling
2018-08-04 18:53:20 [Pool]      [digibyte-qubit] (Thread 1) Block notification via RPC polling
2018-08-04 18:53:24 [Pool]      [digibyte-qubit] (Thread 2) p2p peer node disconnected - attempting reconnection...
2018-08-04 18:53:24 [Pool]      [digibyte-qubit] (Thread 1) p2p peer node disconnected - attempting reconnection...
2018-08-04 18:53:24 [Pool]      [digibyte-qubit] (Thread 2) p2p connection successful
2018-08-04 18:53:24 [Pool]      [digibyte-qubit] (Thread 1) p2p connection successful

---digibyte getpeerinfo---
 {
    "id": 56,
    "addr": "127.0.0.1:53512",
    "addrbind": "127.0.0.1:12024",
    "services": "0000000000000001",
    "relaytxes": true,
    "lastsend": 1533408926,
    "lastrecv": 1533408926,
    "bytessent": 151,
    "bytesrecv": 123,
    "conntime": 1533408926,
    "timeoffset": 0,
    "version": 70016,
    "subver": "/node-stratum/",
    "inbound": true,
    "addnode": false,
    "startingheight": 0,
    "banscore": 0,
    "synced_headers": -1,
    "synced_blocks": -1,
    "inflight": [
    ],
    "whitelisted": false,
    "bytessent_per_msg": {
      "verack": 24,
      "version": 127
    },
    "bytesrecv_per_msg": {
      "version": 123
    }
  },

**Comment by leshacat** [comment]
I'll take a peek when I get back. Maybe somewhere in init.js

I would suggest posting this again on Foxer666's fork, I think Zone117 hasn't touched this in years.

**Comment by leshacat** [comment]
Have you or are you going to repost this on Foxer666 repo? Could you post a link here and close it if so?

Also make sure to post in the right repository, either node-open-mining-portal, stratum-pool, node-multi-hashing, multi-hashing, etc...

I will look at it in the meantime.

User suggested fix may possibly be in https://github.com/bitpay/bitcore-p2p/blob/master/lib/peer.js

Thanks

**Comment by leshacat** [comment]
Not related to that issue, but probably related to these:
https://github.com/bitcoin/bitcoin/issues/9540
https://bitcoin.stackexchange.com/questions/52343/how-do-i-detect-a-ddos-attack-on-my-node

Read and follow those.

TL;DR; set peerbloomfilters=1 in all of your wallet.conf and restart nodes

**Comment by leshacat** [comment]
**Also related:**

https://github.com/bitcoin/bitcoin/pull/7708
https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/528

**Comment by leshacat** [comment]
Turns out the server restarting is due to the node-multihashing issue here: https://github.com/foxer666/node-stratum-pool/pull/33

The fix is here https://github.com/1301313Y/BootNOMP/pull/2

It is tested on my pool here: http://easyxpool.tk:44420/

This issue is fixed in my NOMP repository: https://github.com/leshacat/BootNOMP
This is also fixed in node-multi-hashing if you prefer: https://github.com/leshacat/node-multi-hashing

The person connecting to the port and sending garbage is still an issue, however it ***does not*** restart the server.

Still recommend a fix: disconnecting / banning IP's which send garbage (anything not stratum protocol)

**Comment by zone117x** [comment]
Are there any pull request to get that fixed here?

**Comment by leshacat** [comment]
@zone117x https://github.com/1301313Y/BootNOMP/pull/2

I will submit a push - I patched the multi-hashing and I forgot I had to fork it off you so I can submit it

Few minutes I repost

**Comment by leshacat** [comment]
@zone117x https://github.com/foxer666/node-stratum-pool/pull/34

That is my merge request to patch this issue https://github.com/1301313Y/BootNOMP/pull/2

**Comment by minkcrypto** [comment]
@zone117x This pr fixes this issue https://github.com/zone117x/node-stratum-pool/pull/177
